controllers/upgrade.go

```diff
@@ -634,3 +634,29 @@ func (r *RollingUpgradeContext) endTimeUpdate() {
 		common.TotalProcessingTime(r.RollingUpgrade.ScalingGroupName(), totalProcessingTime)
 	}
 }
+
+// AWS API call for setting an instance to StandBy has a limit of 19. Hence we have to call the API in batches.
+func (r *RollingUpgradeContext) SetBatchStandBy(instanceIDs []string) error {
+	var (
+		pendingInstances = len(instanceIDs)
+		numberOfBatches  = math.Ceil(float64(len(instanceIDs)) / float64(awsprovider.InstanceStandByLimit))
+		start            = 0
+	)
+
+	fmt.Println("SetBatchStandBy", "numberOfBatches", numberOfBatches, "pendingInstances", pendingInstances)
+	for i := 0; i < int(numberOfBatches) && pendingInstances > 0 && start < len(instanceIDs); i++ {
```

Comment:
I'm wondering if this would be more efficient to chunk the slice  first then loop instead of keeping multiple counters in a loop?


Ex.
```
func chunkSlice(slice []int, chunkSize int) [][]int {
	var chunks [][]int
	for i := 0; i < len(slice); i += chunkSize {
		end := i + chunkSize

		// necessary check to avoid slicing beyond
		// slice capacity
		if end > len(slice) {
			end = len(slice)
		}

		chunks = append(chunks, slice[i:end])
	}

	return chunks
}

```